src/main/java/com/aws/greengrass/shadowmanager/ShadowManager.java

```diff
@@ -91,6 +88,7 @@ public class ShadowManager extends PluginService {
     private final AuthorizationHandlerWrapper authorizationHandlerWrapper;
     private final InboundRateLimiter inboundRateLimiter;
     private final DeviceConfiguration deviceConfiguration;
+    private ComponentConfiguration componentConfiguration;
```

Comment:
how is this being used? You are updating this field but never reading from it.

Reply:
This is used in L263 `from` where the last updated value of `componentConfiguration` is used as the old state. For the first time, it is null.